solidity/dashboard/src/sagas/coverage-pool.js

```diff
@@ -174,21 +213,97 @@ export function* subscribeToCovTokenTransferEvent() {
     const tvlInUSD = keepInUSD.multipliedBy(KEEP.toTokenUnit(tvl)).toFormat(2)
     const apy = yield call(Keep.coveragePoolV1.apy)
 
+    const pendingWithdrawal = yield call(
+      Keep.coveragePoolV1.pendingWithdrawal,
+      address
+    )
+
+    const withdrawalInitiatedTimestamp = yield call(
+      Keep.coveragePoolV1.withdrawalInitiatedTimestamp,
+      address
+    )
+
+    const covTokensAvailableToWithdraw = sub(updatedCovBalance, pendingWithdrawal)
+
     yield put(
       covTokenUpdated({
         covBalance: updatedCovBalance,
+        covTokensAvailableToWithdraw,
         covTotalSupply: updatedCovTotalSupply,
         shareOfPool,
         estimatedKeepBalance,
         estimatedRewards,
         totalValueLocked: tvl,
         totalValueLockedInUSD: tvlInUSD,
         apy,
+        pendingWithdrawal,
+        withdrawalInitiatedTimestamp,
       })
     )
   }
 }
 
+export function* subscribeToWithdrawalInitiatedEvent() {
+  const requestChan = yield actionChannel(
+    COVERAGE_POOL_WITHDRAWAL_INITIATED_EVENT_EMITTED
+  )
+
+  while (true) {
+    const {
+      payload: { event },
+    } = yield take(requestChan)
+    const {
+      returnValues: { underwriter, covAmount },
+    } = event
+
+
+    const address = yield select(selectors.getUserAddress)
+
+    if (isSameEthAddress(address, underwriter)) {
+      yield put({
+        type: "modal/is_opened",
+        payload: {
+          transactionHash: event.transactionHash,
+          additionalData: {
+            covAmount
+          }
+        },
+      })
+    }
+  }
+}
+
+export function* subscribeToWithdrawalCompletedEvent() {
+  const requestChan = yield actionChannel(
+    COVERAGE_POOL_WITHDRAWAL_COMPLETED_EVENT_EMITTED
+  )
+
+  while (true) {
+    const {
+      payload: { event },
+    } = yield take(requestChan)
+    const {
+      returnValues: { underwriter, amount },
+    } = event
+    const address = yield select(selectors.getUserAddress)
+
+    if (isSameEthAddress(address, underwriter)) {
+      yield put({
+        type: "modal/is_opened",
+        payload: {
+          emittedEvent: EVENTS.COVERAGE_POOLS.WITHDRAWAL_COMPLETED,
+          transactionHash: event.transactionHash,
+          additionalData: {
+            amount
+          }
```

Comment:
Let's try to achieve the same approach as we use to display [messages](https://github.com/keep-network/keep-core/blob/main/solidity/dashboard/src/sagas/staking.js#L289-L296).

Reply:
Fixed in cd9abad98ab755f57b1cd5ff39a2afce4545fe20